modules/benchmark/R/clone_run.R

```diff
@@ -0,0 +1,115 @@
+##-----------------------------------------------------------------------------#
+##' Produce a "cloned" run from the settings of a completed run
+##' 
+##' @name clone_run 
+##' @title Clone Run
+##' @param settings settings or multisettings object 
+##' @param model model id 
+##' @param con database connection
+##' @export 
+##' 
+##' @author Betsy Cowdery 
+##' @importFrom dplyr tbl filter rename collect select 
+
+clone_run <- function(settings, model, con){
+  
+  # For testing: 
+  # bety <- betyConnect("web/config.php")
+  # con <- bety$con
+  # settings <- read.settings("/fs/data2/output/PEcAn_1000008371/pecan.CHECKED.xml")
+  # model = 1000000020
+  
+  settings.old <- settings 
+  settings.new <- list()
+  
+  # Copy over settings that won't change
+  settings.new$meta.analysis <- settings.old$meta.analysis
+  settings.new$ensemble <- settings.old$ensemble
+  settings.new$run$site <- settings.old$run$site
+  settings.new$run$start.date <- settings.old$run$start.date
+  settings.new$run$end.date <- settings.old$run$end.date
+  
```

Comment:
I tried it the other way and ended up changing my mind - I'll try to explain my reasoning. The point of this function is to create a `pecan.xml`, not a `pecan.CHECKED.xml` which just means that a lot of the information that's imported from the old settings really doesn't need to be copied over because it will be automatically generated when the suit of settings functions are run (like `prepare.settings` and `check.settings`). This covers things like  `info`, `host`, `database`, `rundir`, `modeloutdir`.
Also, in some cases, like `pfts`, all the possible sub tags are filled in `pecan.CHECKED.xml` but one only needs to specify the id and the rest of the information is filled in automatically. The list of things to delete seems longer than what needs to be saved at this point. 
So we might as well start with a clean palate and only copy over the bare minimum of information and let everything else get filled in automatically, reducing the chance of errors/conflicts. 

Alternatively, we could take the approach of moving everything over and "commenting out" sections that are problematic - but I'm still not clear on how to do that well when writing to xml from a list ... I'd just rather all that other information were in a log file as we discussed.

Reply:
@bcow I strongly disagree with your logic here. By my count we have at least 83 xml tags defined in our documentation, and this was without anything from benchmarking (which is missing from the documentation) and the link to the PDA tags broken (@tonygardella please fix this). There are inevitably additional tags that are undocumented and more important there are new tags added constantly. The idea of being able to construct a settings from scratch is simply unsustainable — it would be too hard to ensure the new settings is constructed correctly and not missing anything. Furthermore, only a handful of those tags would need to be changed, most will be identical, so why not just copy the whole object?

I also really don’t follow why you think you can delete large sections. A pecan.xml will not run without host, database, etc and I don’t see where these come from. I also don’t see why you want to loose that info as there may be custom settings that need to be preserved and provenance that needs to be retained and augmented (ie info tag) rather than deleted. 

Finally, I don’t understand what you think needs to be commented out

Reply:
Point taken about new tags being added regularly. I agree my approach doesn't make sense if we want to be able to keep on top of continuing changes. 

Tags like host and database can get filled in using the config list that is generated from config.php. I assumed we would want to delete them because the new cloned run could be on a new server. Those would be filled in anyway if the old settings had already been cleaned / were from a reference run. What kind of custom settings should I be looking out for? 

My thought on commenting things out was half baked and doesn't even really make sense to me either.